### PR TITLE
feat: add Conditions of Joining status to PMs

### PR DESCRIPTION
**Pull request by Judge40**
Add a Conditions of Joining status to Programme Membership panels, show the signed date and Gold Guide version when signed. Otherwise show a status message dependant on the start date of the programme.

TIS21-4372

### CODE REVIEW

**Comment by john12321** [comment]
> Quality analysis was skipped as no source changes were detected.

That's strange

**Review by john12321 — DISMISSED** [review]
LGTM even with no source changes! 😂

**Comment by Judge40** [comment]
> LGTM even with no source changes! 😂

I've raised #841 to fix that, other PRs have had cypress changes bundled in which covered up the `src` folder change.

File: components/common/PanelsCreator.tsx [hunk 1]
```diff
@@ -49,6 +50,11 @@ export function PanelsCreator({
                           <Curricula
                             curricula={filteredPanel[panelProp] as Curriculum[]}
                           />
+                        ) : panelProp === "conditionsOfJoining" ? (
+                          <ConditionsOfJoiningField
+                            conditionsOfJoining={filteredPanel[panelProp]}
+                            startDate={filteredPanel["startDate"]}
+                          />
                         ) : (
                           displayListVal(filteredPanel[panelProp], panelProp)
```
Comment (john12321) [inline components/common/PanelsCreator.tsx:59]: We can probably leave this nested ternary refactoring until your Coj status and @EdwardBarclay's Coj link are combined into a single Coj comp?

**Review by john12321 — APPROVED** [review]

**Pull request merged**